### ISSUE TITLE
Bing Maps Search

### ISSUE THREAD

**Issue by nlcano** [infusion-code/angular-maps#46]
Hi there,

Do you have any sample with Bing Maps and the SearchManager  (Microsoft.Maps.Search) implemented ? 

Thanks for your help!
-NC

**Comment by thor-schueler** [comment]
@nlcano, no we currently do not have support for this in place as part of angular-maps. We do have support for loading modules etc. But you'd have to implement the payloads yourself and then bind the results to the various components.

**Comment by nlcano** [comment]
@thor-schueler: Thanks, I am planning to add support for the Search Feature (and others).

The code listed below works when I use the published package.   I clone and link the package locally, and I am getting the issue below, any help will be appreciated.   

**Module**
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { Routes, RouterModule } from '@angular/router';
import { MapModule, MapAPILoader, WindowRef, DocumentRef, BingMapAPILoaderConfig, BingMapAPILoader } from 'angular-maps';
import { environment } from 'environments/environment';

import { BMapDemoComponent } from './bmapdemo.component';

const routes: Routes = [{
	path: '',
	data: {
      title: 'Maps Tests',
      urls: [{ title: 'Test', url: '/' }, { title: 'Tests' }, { title: 'Bing Maps' }]
    },
    component: BMapDemoComponent
}];

@NgModule({
	imports: [
        CommonModule,
        MapModule.forRootBing(),
        RouterModule.forChild(routes)
    ],
    declarations: [BMapDemoComponent],
    providers: [
        {
            provide: MapAPILoader, deps: [], useFactory: BingMapServiceProviderFactory
        }
    ],
})
export class BMapDemoModule { }

// Bing Maps
export function BingMapServiceProviderFactory() {
    const bc: BingMapAPILoaderConfig = new BingMapAPILoaderConfig();
    bc.apiKey = environment.Bing.Maps.Key;
    bc.branch = 'experimental';
    return new BingMapAPILoader(bc, new WindowRef(), new DocumentRef());
}

**Error:**
Error: Uncaught (in promise): Error: Unexpected value '[object Object]' imported by the module 'BMapDemoModule'. Please add a @NgModule annotation.
Error: Unexpected value '[object Object]' imported by the module 'BMapDemoModule'. Please add a @NgModule annotation.
    at syntaxError (compiler.js:466)
    at eval (compiler.js:15088)
    at Array.forEach (<anonymous>)
    at CompileMetadataResolver.getNgModuleMetadata (compiler.js:15071)
    at JitCompiler._loadModules (compiler.js:33486)
    at JitCompiler._compileModuleAndComponents (compiler.js:33447)
    at JitCompiler.compileModuleAsync (compiler.js:33363)
    at CompilerImpl.compileModuleAsync (platform-browser-dynamic.js:230)
    at eval (core.js:6441)
    at ZoneDelegate.invoke (zone.js:392)
    at syntaxError (compiler.js:466)
    at eval (compiler.js:15088)
    at Array.forEach (<anonymous>)
    at CompileMetadataResolver.getNgModuleMetadata (compiler.js:15071)
    at JitCompiler._loadModules (compiler.js:33486)
    at JitCompiler._compileModuleAndComponents (compiler.js:33447)
    at JitCompiler.compileModuleAsync (compiler.js:33363)
    at CompilerImpl.compileModuleAsync (platform-browser-dynamic.js:230)
    at eval (core.js:6441)
    at ZoneDelegate.invoke (zone.js:392)
    at resolvePromise (zone.js:824)
    at resolvePromise (zone.js:795)
    at eval (zone.js:873)
    at ZoneDelegate.invokeTask (zone.js:425)
    at Object.onInvokeTask (core.js:4620)
    at ZoneDelegate.invokeTask (zone.js:424)
    at Zone.runTask (zone.js:192)
    at drainMicroTaskQueue (zone.js:602)
    at <anonymous>

**Comment by nlcano** [comment]
@thor-schueler: Thanks, I am planning to add support for the Search Feature (and others).

I clone and link the package locally, and I am getting the issue _Uncaught (in promise): Error: Unexpected value '[object Object]', any help/guidance will be appreciated.

**Comment by thor-schueler** [comment]
@nlcano: yeah, linking is tricky. When you link, the entire project gets linked into the node_modules folder of you local project, including the .ts files. So when you build, the ngc compiler will try to compile those file as well (depending on how your tsconfig in the local test project is setup). However, all the references will be screwed up, hence the errors (at least this is what I suspect). 

You really have two options here:

- build the package in your local clone (npm run prepublish) and then copy the contents of the dist folder into the appropriate place in the node_modules of your local test project (instead of using npm link).  That is what I usually do. Since the dist won't contain the ts files, you'll be in good shape. 
- adjust the tsconfig file in your local test project to exclude any .ts files in node_modules from the compile. 

If you continue to have issues getting it setup, let me know and I'll setup a skype to look at it with you. Once you got a local version, adding the search capabilities will likely include:

- Defining a search service contract. (see the abstracts in the services root as go-bys)
- Implement the contract for bing (and preferably for google). 
- Create a component the renders the search box for inclusion in the map. You'd inject the service into the component and the results would probably be an observable that that map host can subscribe to to do appropriate things. 

Hope that helps....

**Comment by abhishekgupta1991** [comment]
Looking for only search box without bing map like autocomplete. When we are trying write in the box will hit the bing api and get the results. If you can advice, how we can achieve it with angular map.
